src/Npgsql/NpgsqlConnector.Auth.cs

```diff
@@ -246,58 +253,76 @@ static byte[] Xor(byte[] buffer1, byte[] buffer2)
                 return buffer1;
             }
 
-            static byte[] HMAC(byte[] data, string key)
+            static byte[] HMAC(byte[] key, string data)
             {
-                using var hmacsha256 = new HMACSHA256(data);
-                return hmacsha256.ComputeHash(Encoding.UTF8.GetBytes(key));
+                using var hmacsha256 = new HMACSHA256(key);
+                return hmacsha256.ComputeHash(Encoding.UTF8.GetBytes(data));
             }
         }
 
-        async Task AuthenticateMD5(string username, byte[] salt, bool async, CancellationToken cancellationToken = default)
+        async Task AuthenticateMD5(string username, int salt, bool async, CancellationToken cancellationToken = default)
         {
             var passwd = GetPassword(username);
             if (passwd == null)
                 throw new NpgsqlException("No password has been provided but the backend requires one (in MD5)");
 
-            byte[] result;
+            const int HexHashSize = 2 * 128 / 8;
+
+            var hashLength = HexHashSize + 3 + 1;
+            var hash = ArrayPool<byte>.Shared.Rent(hashLength);
+
             using (var md5 = MD5.Create())
```

Comment:
In case of dropping .NS 2.0 we use the (new) static method `MD5.HashData`? 
This one has span-overloads too.

Reply:
Supporting .NET Standard blocked me from doing this and for the same reason I used pools instead of spans.